test/git-unix/test_rev_list.ml

```diff
@@ -0,0 +1,139 @@
+
+  let ( >>?= ) = Lwt_result.bind
+
+  let consume stream =
+    let rec go () =
+      stream () >>= function
+      | Some _ -> go ()
+      | None -> Lwt.return () in
+    go ()
+
+  let rev_list t fake_remote_references fake_commands : (Store.Hash.t list, Store.error) result Lwt.t =
+    Sync.packer t ~ofs_delta:true fake_remote_references fake_commands
+    >>?= (fun (stream, idx) ->
+      consume stream
+      >>= fun () -> Lwt_mvar.take idx
+      >|= fun graph -> Store.Pack.Graph.fold (fun hash _ acc -> hash :: acc) graph []
+        |> fun v -> Ok v)
+
+  let rev_list t cs =
+    let fake_remote_references =
+      List.fold_left
+        (fun acc -> function
+          | { fake_remote_references
+            ; _ } -> List.fold_left (fun acc (hash, reference, peeled) -> Store.Reference.Map.add (Store.Reference.of_string reference) (hash, peeled) acc) acc fake_remote_references)
+        Store.Reference.Map.empty cs
+      |> Store.Reference.Map.bindings
+      |> List.map (fun (reference, (hash, peeled)) -> (hash, Store.Reference.to_string reference, peeled)) in
+    let module OrderedCommand = struct type t = Sync.command let compare a b = match a, b with
+                                         | `Create (a, _), `Create (b, _) -> Store.Hash.compare a b
+                                         | `Delete (a, _), `Delete (b, _) -> Store.Hash.compare a b
+                                         | `Update (a, b, _), `Update (c, d, _) ->
+                                            let ret = Store.Hash.compare a c in
+                                            if ret = 0 then Store.Hash.compare b d else ret
+                                         | `Create _, _ -> 1
+                                         | `Delete _, `Create _ -> (-1)
+                                         | `Delete _, _ -> 1
+                                         | `Update _, `Create _ -> (-1)
+                                         | `Update _, `Delete _ -> (-1) end in
+    let module Set = Set.Make(OrderedCommand) in
+    let fake_commands =
+      List.fold_left (fun acc { fake_commands; _ } -> List.fold_left (fun acc x -> Set.add x acc) acc fake_commands)
+        Set.empty cs
+      |> Set.elements in
+    rev_list t fake_remote_references fake_commands
+
+  let input_rev_list = function
+    | { delta = `Diff (from_, to_); _ } ->
+       [ `Negative from_; `Object to_ ]
+    | { delta = `New hash; _ } ->
+       [ `Object hash ]
+
+  let git_rev_list cs =
+    let command = Fmt.strf "git rev-list --objects --stdin" in
+    let input = fun oc ->
+      let ppf = Format.formatter_of_out_channel oc in
+      let pp_elt ppf = function
+        | `Negative hash -> Fmt.pf ppf "^%a" Store.Hash.pp hash
+        | `Object hash -> Store.Hash.pp ppf hash in
+      Fmt.(pf ppf) "%a\n%!" Fmt.(list ~sep:(const string "\n") pp_elt) (List.concat (List.map input_rev_list cs)) in
+    let git_dir = Fpath.(root / ".git") in
```

Comment:
Can't you pass `--git-dir <dir>` to your command instead of tweaking the environment?